src/main/java/frc/robot/subsystems/swerve/PhoenixOdometryThread.java

```diff
@@ -51,22 +53,40 @@ public static PhoenixOdometryThread getInstance() {
   private PhoenixOdometryThread() {
     setName("PhoenixOdometryThread");
     setDaemon(true);
-    start();
+  }
+
+  @Override
+  public void start() {
+    if (timestampQueues.size() > 0) {
+      super.start();
+    }
   }
 
   public Queue<Double> registerSignal(ParentDevice device, StatusSignal<Double> signal) {
-    isCANFD = CANBus.isNetworkFD(device.getNetwork());
-    Queue<Double> queue = new ArrayBlockingQueue<>(100);
+    Queue<Double> queue = new ArrayDeque<>(100);
     signalsLock.lock();
+    SwerveSubsystem.odometryLock.lock();
```

Comment:
Activating a lock in an entirely different class is a violation of separation of concerns and should be avoided. It also produces potential for deadlock. Why does this need to happen?

Reply:
I'm not sure, where would I move it to in SwerveSubsystem?